spec/services/target_matching_service_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe TargetMatchingService do
+  let!(:user) { create(:user) }
+  let(:topic) { target.topic_id }
```

Comment:
I would define this the other way round. 
`let!(:topic) { create(:topic) }`
And then use the `topic.id` where it corresponds.

Maybe you can also define here 2 topics to use them in the different targets.
`let!(:topic_a) { create(:topic) }`
`let!(:topic_b) { create(:topic) }`